packages/at_chops/lib/src/at_chops_base.dart

```diff
@@ -0,0 +1,69 @@
+import 'dart:typed_data';
+
+import 'package:at_chops/src/algorithm/at_algorithm.dart';
+import 'package:at_chops/src/algorithm/default_encryption_algo.dart';
+import 'package:at_chops/src/algorithm/default_hashing_algo.dart';
+import 'package:at_chops/src/algorithm/default_signing_algo.dart';
+import 'package:at_chops/src/algorithm/pkam_signing_algo.dart';
+import 'package:at_chops/src/key/impl/at_chops_keys.dart';
+import 'package:at_chops/src/key/key_type.dart';
+
+/// Base class for all Cryptographic and Hashing Operations. Callers have to either implement
+/// specific encryption, signing or hashing algorithms. Otherwise default implementation of specific algorithms will be used.
+abstract class AtChops {
+  final AtChopsKeys _atChopsKeys;
+
+  AtChopsKeys get atChopsKeys => _atChopsKeys;
+
+  AtChops(this._atChopsKeys);
+
+  /// Encrypts the input bytes [data] using an [encryptionAlgorithm].
+  /// If [encryptionKeyType] is [EncryptionKeyType.rsa_2048] then [encryptionAlgorithm] will be set to [DefaultEncryptionAlgo]
+  Uint8List encryptBytes(Uint8List data, EncryptionKeyType encryptionKeyType,
```

Comment:
As discussed previously, please can you add a 'keyName' parameter to the various asymmetric encrypt and decrypt methods which defaults to null which means legacy behaviour. 

In future we may have multiple asymmetric keypairs and we need a way to distinguish between them.

Reply:
multiple asymmetric keypairs - meaning different keys types e.g rsa 2048, rsa 4096 or different keys for the same key type e.g two key pairs generated for rsa 2048 ?

Reply:
Both

Reply:
We just need to be able to identify them. What I have in mind is that names will be derivable from the names of their entries in the keystore, and vice versa. For example we may store `@alice`'s public keys at
```
rsa1.__public_keys@alice
1234567.__public_keys@alice
ecc_67890_abc.__public_keys@alice
```
and then the `keyName` for each of those would be the first part i.e. 
```
rsa1
1234567
ecc_67890_abc
```

Reply:
done